spec/unit/cli/cli_spec.rb

```diff
@@ -0,0 +1,120 @@
+
+      it 'should work with arrays' do
+        options = {path: 'foo.bar.array'}
+        expected = "[\n    1,\n    2,\n    3\n]"
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq(expected)
+      end
+
+      it 'should work with hashes' do
+        options = {path: 'foo.bar.hash'}
+        expected = "key: value\n"
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq(expected)
+      end
+
+      it 'should output json if specified' do
+        options = {path: 'foo.bar.hash', json: true}
+
+        # Note that this is valid json, while the test above is not
+        expected = "{\n    \"key\": \"value\"\n}\n"
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq(expected)
+      end
+    end
+
+    context 'do_set()' do
+      it 'should overwrite values' do
+        options = {path: 'foo.bar.baz', new_value: 'pi'}
+        expected = hocon_text.sub(/42/, 'pi')
+        expect(Hocon::CLI.do_set(options, hocon_text)).to eq(expected)
+      end
+
+      it 'should create new nested values' do
+        options = {path: 'new.nested.path', new_value: 'hello'}
+        expected = "new: {\n  nested: {\n    path: hello\n  }\n}"
+        # No config is supplied, so it will need to add new nested hashes
+        expect(Hocon::CLI.do_set(options, '')).to eq(expected)
+      end
+
+      it 'should allow arrays to be set' do
+        options = {path: 'my_array', new_value: '[1, 2, 3]'}
+        expected = 'my_array: [1, 2, 3]'
+        expect(Hocon::CLI.do_set(options, '')).to eq(expected)
+      end
+
+      it 'should allow arrays in strings to be set as strings' do
+        options = {path: 'my_array', new_value: '"[1, 2, 3]"'}
+        expected = 'my_array: "[1, 2, 3]"'
+        expect(Hocon::CLI.do_set(options, '')).to eq(expected)
+      end
+
+      it 'should allow hashes to be set' do
+        do_set_options = {path: 'my_hash', new_value: '{key: value}'}
+        do_set_expected = 'my_hash: {key: value}'
+        do_set_result = Hocon::CLI.do_set(do_set_options, '')
+        expect(do_set_result).to eq(do_set_expected)
+
+        # Make sure it can be parsed again and be seen as a real hash
+        do_get_options = {path: 'my_hash.key'}
+        do_get_expected = 'value'
+        expect(Hocon::CLI.do_get(do_get_options, do_set_result)).to eq(do_get_expected)
+      end
+
+      it 'should allow hashes to be set as strings' do
+        do_set_options = {path: 'my_hash', new_value: '"{key: value}"'}
+        do_set_expected = 'my_hash: "{key: value}"'
+        do_set_result = Hocon::CLI.do_set(do_set_options, '')
+        expect(do_set_result).to eq(do_set_expected)
+
+        # Make sure it can't be parsed again and be seen as a real hash
+        do_get_options = {path: 'my_hash.key'}
+        expect{Hocon::CLI.do_get(do_get_options, do_set_result)}
+            .to raise_error(Hocon::ConfigError::ConfigWrongTypeError)
+      end
+    end
+
+    context 'do_unset()' do
+      it 'should remove values' do
```

Comment:
Do we need a way to ask if a value exists?  Not sure the use cases, but I'd consider it.

For example, what if we only wanted to change something if it exists?